src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { useState, useEffect } from "react";
+import { Button, Container, Form, Row, Col, } from "react-bootstrap";
+import { MovieCard } from "../movie-card/movie-card";
+import { FavoriteMovies } from "./favorite-movies";
+
+
+export const ProfileView = ({ user, movies }) => {
+    const storedToken = localStorage.getItem("token");
+    const storedMovies = localStorage.getItem("movies");
+    const [allMovies] = useState(storedMovies ? storedMovies: movies);
+    const [token] = useState(storedToken ? storedToken : null);
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [email, setEmail] = useState('');
+    //const [birthday, setBirthday] = useState('');
+   // const [favorites, setFavorites] = useState('');
+    const storedFav = localStorage.getItem("favorites");
+    const storedUser = localStorage.getItem("user");
+    // const [user, setUser] = useState(storedUser ? storedUser : null);
+   // const {Username, Birthday, Email, Favorites} = user;    
+   // const [userFavoriteMovies] = useState(storedUser.Favorites ? storedUser.Favorites: Favorites);
+    const [filteredMovies, setFilteredMovies] = useState(allMovies);
+    //const [selectedId, setSelectedId] = useState("");
+    console.log(user, storedFav);
+  //  console.log("filteredMovies: ", filteredMovies)
+    let hasMovieId = localStorage.getItem("favorites");
+
+
+
+
+     const favoriteMovieList = movies.filter((m) => {
+              /* console.log("movie: ", m.id,"-", m.title);
+              console.log("favorite ?: ", storedFav.includes(m.id), storedFav.indexOf(m.id)); */
+           return storedFav.includes(m.id);
+           /*storedFav.includes(m.id);*/
+
+
+          
+    });
+   
+
+
+    /*<div id="car-list">
+    {filteredMovies.map((item, index) => (
+       <div className="car-item" key={index}>
+         <div className="car-name">{`Name: ${item.name}`}</div>
+         <div className="car-year">{`Year: ${item.release_year}`}</div>
+         <img className="car-image" src={item.url} alt="car-img" />
+       </div>
+       ))}
+       </div>*/
+     
+   // console.log("favoriteMovies", favoriteMovies);
+
+
+   
+    const updateUser = (user) => {
+        fetch("https://myflix-api-3dxz.onrender.com/users/"+user, {
+          headers: { Authorization: `Bearer ${token}` },
+        })
+          .then((response) => response.json())
+          .then((data) => {
+            setUser(data.user);
+            localStorage.setItem("user", JSON.stringify(data.user));
+          //  console.log("User->", user)
+          });
+    };
```

Comment:
IMO this should be inside a `useEffect` hook since it first gets the user data used to pre-fill the form. Please read on useEffect hook and how to use it for side effects
```suggestion
    useEffect(() => {
    const updateUser = (userId) => {
        fetch("https://myflix-api-3dxz.onrender.com/users/"+userId, {
          headers: { Authorization: `Bearer ${token}` },
        })
          .then((response) => response.json())
          .then((data) => {
            setInputValues(data.user);
            localStorage.setItem("user", JSON.stringify(data.user));
          });
    };
    updateUser(userId);
  }, [userId]);
```